libbeat/common/file/rotator.go

```diff
@@ -310,7 +310,7 @@ func (r *Rotator) purgeOldBackups() error {
 }
 
 func (r *Rotator) purgeOldIntervalBackups() error {
-	files, err := filepath.Glob(r.filename + "*")
```

Comment:
what happens to the other files? like `filebeat.3`? are they ignored?

Reply:
Yes. They will not be part of the rotation

Reply:
this fixes an issue with a rotation.
but also can you make sure other files from other rotator are removed as well?
there are multiple strategies how to accomplish this and i'm not sure which one to use

1. keep them there and wait to be cleaned manually
1. on `New` remove backups of opposite rotator 
1. on rotate - rotate in a way that number of log files will be r.max consisting of mixture of previous and current logs sorted correctly (preference of current). so if i have `max==10` and 7 files from current rotator and 3 of previous one, on rotation i will end up with `8+2` in the next step `9+1` and finally with clean `10+0`
1. on rotate find out `modTime` of each file, sort by `modtime` remove all but `r.max` newest

i'm also ok with doing this in a followup as this PR already fixes an issue with missing logs.

Reply:
thinking more about it we have another most valid option here:
- let's get rid of `filename.X` format
- in this case we don't need to pay attention to file format during rotation
- with new format in mind, we need to transform old names to new ones on startup (if any) `filename.X` => `filename.{file.ModTime}` 

@leopucci let me know if this is something you would like to cover in this PR or I should create an additional issue for this so somebody can address this in a followup

Reply:
@michalpristas I have created the function to purge the .X filenames on startup. 
Also created a test function. TestIntervalRotationPurgeFileRotationLogs

daily-2009-11-17-
daily-2009-11-18-
daily-2009-11-19-

Names are not ideal, can be better formatted. Please let me know what you think of it
Regards